virtool/tasks/runner.py

```diff
@@ -19,24 +19,42 @@ def __init__(
         self._data = data
         self._tasks_client = tasks_client
         self.app = app
+        self.current_task = None
 
     async def run(self):
-        logging.info("Started task runner")
+        task_id = None
 
         try:
             while True:
                 logging.info("Waiting for next task")
-
                 task_id = await self._tasks_client.pop()
 
                 await self.run_task(task_id)
 
+                await asyncio.shield(self.current_task)
+
                 logging.info("Finished task: %s", task_id)
 
         except asyncio.CancelledError:
-            logging.info("Stopped task runner")
+            if task_id is not None:
+                logging.info("Recieved stop signal; awaiting task completion")
```

Comment:
This should maybe be two logging calls.

1. Received stop signal where _Stopped task runner_ used to be. It is logged whether or not a task is active.
2. _Waiting for running task to finish/complete_ if there is a current task active.